docs/e2e/extension-e2e-guidelines.md

```diff
@@ -314,3 +315,49 @@ Test Name: Connects to a Hardware wallet for Trezor
 ```
 
 ✅ Proposed solution: The Trezor import flow involves opening the Trezor website, then the user takes additional steps on that website to connect the device. A fake version of this website could be created for testing, and the test build could be updated to use this fake version. It's also worth investigating a phishing detection solution, such as replacing Github.com with an empty page.
+
+## Page Object Model (POM)
+
+POM, or Page Object Model, is a design pattern commonly utilized for writing automated end-to-end tests. A page object is a instance of a class that acts as an interface for the page of the application under test.
+This design pattern is being adopted due to its numerous benefits, including improved test maintenance, increased code reusability, and enhanced readability of test scripts. By abstracting the UI details into page objects, changes to the application's UI require updates only in the page objects, significantly reducing the effort needed to maintain tests as the application evolves. The adoption of POM in our e2e testing strategy aims to leverage these advantages to create a more robust, maintainable, and efficient testing framework.
+
+### Composition of a Page Class
+
+Each page class is composed of:
+
+- **Locators**: HTML elements.
+- **Action Methods**: Methods for interacting with the elements.
+- **Check Methods**: Methods that assert the status of elements.
+
+A test creates page objects and interacts with web elements by calling methods of those page objects.
+
+### Best Practices
+
+- A page object should represent meaningful elements of a page and not necessarily a complete page. It can represent a component of a page, like a navbar. See example [here.](https://github.com/MetaMask/contributor-docs/examples/extension-e2e-page-object-model/home-page.ts#L6)
```

Comment:
Could we include this link https://github.com/MetaMask/metamask-extension/blob/f9f3cff8a8a920f2607c3cd2f1c2e134ff1e06e0/test/e2e/page-objects/pages/homepage.ts#L25

Reply:
i got your point to point to class constructor, so i updated to L23.
but i don't think we should use link like your suggestion to point to a specific PR file, we should point directly to example homepage file in this repo.